vortex-datafusion/src/persistent/opener.rs

```diff
@@ -71,53 +74,39 @@ impl FileOpener for VortexFileOpener {
                 .with_ctx(ctx.clone())
                 .with_file_layout(
                     file_layout_cache
-                        .try_get(&file_meta.object_meta, object_store)
+                        .try_get(&file_meta.object_meta, object_store, io_dispatcher.clone())
                         .await?,
                 )
                 .open()
                 .await?;
 
             // Set up a task executor using the current DataFusion handle to make sure we don't
             // accidentally spawn tasks on the I/O dispatcher.
-            let task_executor = Arc::new(TokioTaskExecutor(Handle::current()));
```

Comment:
I think a lot of the speedd up might be in not spawning all the CPU tasks? Is this related to I/O change?

Reply:
locally it doesn't seem to make a difference, but I'll rerun it on the runners